### PR TITLE
Convert to deployment instead of direct pod

### PR DESCRIPTION
**Pull request by eriknelson**
Still testing this out but I've had pretty reliable syncing of test workloads with it so far using the non-privileged ssh pod.

I would like to follow this up with a PR that removes the requirement for everyone to build their own sshd images with keys baked into them and move to a single centrally published image with an ssh key param.

This PR standardizes the sshd container to ensure that it will run with a non-0 UID that is known to cause breakages under some usage scenarios.

### CODE REVIEW

File: 3_run_rsync/templates/deploy.yml.j2 [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    matchLabels:
+      app: {{ pod_name }}
+  template:
+    metadata:
+      name: {{ pod_name }}
+      namespace: {{ pvc_namespace }}
+      labels:
+        app: {{ pod_name }}
+    spec:
+      imagePullPolicy: Always
```
Comment (eriknelson) [inline 3_run_rsync/templates/deploy.yml.j2:18]: Do we want to add always pull? This has been painful for some people in the past.

Comment (jwmatthews) [reply 3_run_rsync/templates/deploy.yml.j2:18]: I like Always for this

Comment (pranavgaikwad) [reply 3_run_rsync/templates/deploy.yml.j2:18]: +1 for Always

**Review by pranavgaikwad — APPROVED** [review]
ACK

**Comment by jwmatthews** [comment]
Let's hold on this Deployment approach.
The Pod approach might actually be desired for longer term.

**Comment by djwhatle** [comment]
Closing. Re-open if needed.

**Pull request closed without merging**